Prehab2rehab/606 get auditlog for study

iaigner opened the pull request:
Add Auditlog Download to Data Tab:
- update StudyManagerApi.yaml
- include AuditLogAPI & implement corresponding reqeusts
- show information on new Data Tab (Download AuditLog) & provide download button for Study.Admin

Full Task Description:
https://github.com/MORE-Platform/more-studymanager-frontend/issues/606

iaigner committed:
#606: P2R - Add "Download Auditlog" Tab on Monitoring&Data

iaigner committed:
#606: P2R - Update StudyManagerApi.yaml and adapt FE accordingly

iaigner committed:
#606: P2R - fix build problems

iaigner committed:
#606: P2R - fix error with ParticipantDataListener and downloadAuditLog

iaigner committed:
#606: P2R - fix Info Text (if) for auditLogDownload

westei commented on openapi/StudyManagerAPI.yaml:
```diff
@@ -1327,6 +1327,44 @@ paths:
+    get:
+      tags:
+        - auditlog
+      description: Get current length of auditlogs with export format
```
Provides metadata of the AuditLog for the requested study

iaigner replied:
as discussed in descriptions it will be audit logs

westei commented on openapi/StudyManagerAPI.yaml:
```diff
@@ -2135,6 +2174,88 @@ components:
 
+    AuditLog:
+      description: |
+        The auditlog object of one study that includes all Auditlog Entries of said study.
```
Try to use `AuditLog` everywhere

iaigner replied:
refactored everything in the backend and frontend branch. also the functions that were still with auditlog and the operationIds :)

iaigner committed:
#606: P2R - refactor: use "auditLog" instead of "auditlog" & correct descriptions in yaml

iaigner committed:
#606: P2R - refactor: use "auditLog" instead of "auditlog" & correct descriptions in yaml

iaigner committed:
#606: P2R - review adaptations: define content-disposition header in StudyManagerAPI.yaml

iaigner committed:
Merge branch 'develop' into prehab2rehab/606-get-auditlog-for-study

markus-ebner commented on src/stores/studyStore.ts:
```diff
@@ -186,6 +205,15 @@ export const useStudyStore = defineStore('study', () => {
+    auditLogMetadata.value = await auditLogApi.getAuditLogMetadata(studyId)
+      .then((response: AxiosResponse) => response.data)
+
+    console.log('getAuditlogMetadata was called')
```
remove console.log

markus-ebner commented on src/stores/studyStore.ts:
```diff
@@ -170,7 +173,23 @@ export const useStudyStore = defineStore('study', () => {
+    await auditLogApi.exportAuditLog(studyId)
+      .then((response:AxiosResponse<AuditLogEntry[]>) => {
+        window.open(response.headers.location);
+        console.log('response: ', response.data)
```
remove console.log

markus-ebner commented on src/components/subComponents/AuditLogDownload.vue:
```diff
@@ -0,0 +1,61 @@
+  <div>
+    <h3 class="mb-1 font-bold">{{t('data.auditLogDownload.title')}}</h3>
+    <div v-if="!(auditLogMetadata && auditLogMetadata?.length)">{{t('data.auditLogDownload.notStartedInfo')}}</div>
+    <div v-else-if="auditLogMetadata && auditLogMetadata?.length">
```
entweder `auditLogMetadata && auditLogMetadata.length` oder `auditLogMetadata?.length`

markus-ebner replied:
use a computed since, the check is done 3 times

markus-ebner commented on src/stores/studyStore.ts:
```diff
@@ -186,6 +205,15 @@ export const useStudyStore = defineStore('study', () => {
+
+    console.log('getAuditlogMetadata was called')
+
+    return auditLogMetadata.value;
```
do not return the value, since it is exposed via ref - be consistent with delivery in one way.

markus-ebner commented on package-lock.json:
why does your package-lock.json contain so much changes? Haven't we pinned all packages?

iaigner replied:
yes it is pinned. I'll check tomorrow thoroughly what the problem here is.

iaigner replied:
@markus-ebner: i went through the change liste of the package-lock.json. it doesn't seem to be changes with the packages we pinned directly. but sub-packages used by our pinned versions (inside those packages). i did run npm i so my guess it it updated the sub dependencies.

are the changes of the sub-packages problematic?

markus-ebner commented on src/components/ParticipationDataList.vue:
```diff
@@ -612,6 +616,20 @@ Licensed under the Elastic License 2.0. */
+    () => props.pauseDataRefresh,
+    (newVal) => {
+      if (newVal) {
+        console.log('Pause aktiviert:', newVal)
```
remove console.log

markus-ebner commented on src/components/ParticipationDataList.vue:
```diff
@@ -612,6 +616,20 @@ Licensed under the Elastic License 2.0. */
+        console.log('Pause aktiviert:', newVal)
+        clearInterval(timer)
+      } else {
+        console.log('Pause aufgehoben:', newVal)
```
remove console.log

markus-ebner reviewed (commented):
A few small changes needed to be done. 
Please be sharper with the console.logs since this is a occurs every now and then in your PRs. Please review your code changes more thoroughly before sending it out. 
We can also think of adding a new eslint rule, so our pipeline crashes and it is automated.

iaigner committed:
#606: P2R - review adaptations

iaigner committed:
#606: P2R - review adaptations: added eslint no-console-log checks and changed info logouts to console.info + regenerate package.lock.json

iaigner commented:
> A few small changes needed to be done. Please be sharper with the console.logs since this is a occurs every now and then in your PRs. Please review your code changes more thoroughly before sending it out. We can also think of adding a new eslint rule, so our pipeline crashes and it is automated.

@markus-ebner : I added the console.log eslint check -> and changed the logs that logged out information to console.info

iaigner committed:
#606: P2R - review adaptations: rerun npm i due to optional dependencies bug (https://github.com/npm/cli/issues/4828)
